src/vma/dev/ring_simple.cpp

```diff
@@ -777,10 +777,17 @@ bool ring_simple::rx_process_buffer(mem_buf_desc_t* p_rx_wc_buf_desc, void* pv_f
 
 		if (likely((si != NULL) && si->flow_tag_enabled())) { 
 			// will process packets with set flow_tag_id and enabled for the socket
+			if (unlikely(p_eth_h->h_proto == htons(ETH_P_8021Q))) {
+				// Handle VLAN header as next protocol
+				transport_header_len = ETH_VLAN_HDR_LEN;
+			}
```

Comment:
I would remove the initialization of transport_header_len from the top of the function and set it when we know what it's supposed to be. So at the top of the function I'd do:

`size_t transport_header_len;`

And here, I'd add:

```
if (p_eth_h->h_proto == htons(ETH_P_8021Q)) {
        ...
} else {
        transport_header_len = ETH_HDR_LEN;
}
```

You should do exactly the same in the non-flowtag case bellow.